core/http-auth-aws/src/test/java/software/amazon/awssdk/http/auth/aws/internal/signer/DefaultAwsV4HttpSignerTest.java

```diff
@@ -343,4 +344,337 @@ public void sign_WithPayloadSigningNullAndHttp_FallsBackToPayloadSigning() {
+            new TestUtils.AnonymousCredentialsIdentity(),
+            httpRequest -> {
+            },
+            signRequest -> {
+            }
+        );
+
+        AsyncSignedRequest signedRequest = signer.signAsync(request).join();
+        assertThat(signedRequest.request().firstMatchingRawQueryParameter("X-Amz-Signature"))
+            .isNotPresent();
+        assertThat(signedRequest.request().firstMatchingHeader("x-amz-content-sha256")).isNotPresent();
+    }
+
+    @Test
+    public void signAsync_WithQueryAuthLocationAndExpiration_DelegatesToPresignedSigner() {
+        AsyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicAsyncRequest(
+            AwsCredentialsIdentity.create("access", "secret"),
+            httpRequest -> {
+            },
+            signRequest -> signRequest
+                .putProperty(AUTH_LOCATION, AuthLocation.QUERY_STRING)
+                .putProperty(EXPIRATION_DURATION, Duration.ZERO)
+        );
+
+        AsyncSignedRequest signedRequest = signer.signAsync(request).join();
+
+        assertThat(signedRequest.request().firstMatchingRawQueryParameter("X-Amz-Expires")).isPresent();
+    }
+
+    @Test
+    public void signAsync_WithPayloadSigningFalse_DelegatesToUnsignedPayloadSigner() {
+        AsyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicAsyncRequest(
+            AwsCredentialsIdentity.create("access", "secret"),
+            httpRequest -> {
+            },
+            signRequest -> signRequest
+                .putProperty(PAYLOAD_SIGNING_ENABLED, false)
+        );
+
+        AsyncSignedRequest signedRequest = signer.signAsync(request).join();
+
+        assertThat(signedRequest.request().firstMatchingHeader("x-amz-content-sha256")).hasValue("UNSIGNED-PAYLOAD");
+    }
+
+    @Test
+    public void signAsync_WithEventStreamContentTypeWithoutHttpAuthAwsEventStreamModule_throws() {
+        AsyncSignRequest<? extends AwsCredentialsIdentity> request = generateBasicAsyncRequest(
+            AwsCredentialsIdentity.create("access", "secret"),
+            httpRequest -> httpRequest
+                .putHeader("Content-Type", "application/vnd.amazon.eventstream"),
+            signRequest -> {
+            }
+        );
+
+        try (MockedStatic<ClassLoaderHelper> utilities = Mockito.mockStatic(ClassLoaderHelper.class)) {
+            utilities.when(() ->ClassLoaderHelper.loadClass(
+                "software.amazon.awssdk.http.auth.aws.eventstream.HttpAuthAwsEventStream",
+                false)
+            ).thenThrow(new ClassNotFoundException("boom!"));
+            Exception e = assertThrows(RuntimeException.class, () -> signer.signAsync(request).join());
+            assertThat(e).hasMessageContaining("http-auth-aws-eventstream");
+        }
+    }
+
+    @Disabled("Chunk-encoding is not currently supported in the Async signing path - it is handled in HttpChecksumStage for now.")
```

Comment:
hmm, with disabled, I can't tell what the current behavior is. Does it throw exception or does something else. would it make sense to have these tests _document_ the current behavior with TODO explaining the not supported.

Reply:
The test is disabled because it would fail otherwise. It makes the correct assertions.

Reply:
Added comments.